tests/test_transformation.py

```diff
@@ -0,0 +1,290 @@
+                }
+            }
+        },
+
+        "type": "object",
+
+        "properties": {
+            "billing_address": {
+                "type": "object",
+                "properties": {
+                    "street_address": {"type": "string"}
+                }
+            },
+            "shipping_address": {
+                "type": "object",
+                "properties": {
+                    "street_address": {"type": "string"}
+                }
+            }
+        }
+    }
+    result = transform(v1)
+    assert exp == result
+
+
+def test_ref_no_conflict_inside_schema_pass():
+    """Test that having no conflict in schema with references passes."""
+    v1 = {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+
+        "definitions": {
+            "address": {
+                "type": "object",
+                "properties": {
+                    "street_address": {"type": "string"}
+                }
+            }
+        },
+
+        "type": "object",
+
+        "properties": {
+            "billing_address": {"$ref": "#/definitions/address"},
+            "shipping_address": {
+                "allOf": [
+                    {"$ref": "#/definitions/address"},
+                    {
+                        "properties":   {
+                            "type": {"enum": ["residential", "business"]}
+                        }
+                    }
+                ]
+            }
+        }
+    }
+    exp = {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+
+        "definitions": {
+            "address": {
+                "type": "object",
+                "properties": {
+                    "street_address": {"type": "string"}
+                }
+            }
+        },
+
+        "type": "object",
+
+        "properties": {
+            "billing_address": {
+                "type": "object",
+                "properties": {
+                    "street_address": {"type": "string"}
+                }
+            },
+            "shipping_address": {
+                "allOf": [
+                    {
+                        "type": "object",
+                        "properties": {
+                            "street_address": {"type": "string"}
+                        }
+                    },
+                    {
+                        "properties": {
+                            "type": {"enum": ["residential", "business"]}
+                        }
+                    }
+                ]
+            }
+        }
+    }
+    result = transform(v1)
+    assert exp == result
+
+
+def test_ref_conflict_inside_schema():
+    """Test that having an invalid reference in the schema raises
+    RefResolutionError."""
+    v1 = {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+
+        "type": "object",
+
+        "properties": {
+            "billing_address": {"$ref": "#/definitions/differentaddress"},
```

Comment:
The test could be simpler. It seems to me that `"billing_address": {"$ref": "#/definitions/differentaddress"},` is enough to raise the exception. No need for `"shipping_address"`. Or am I missing something?